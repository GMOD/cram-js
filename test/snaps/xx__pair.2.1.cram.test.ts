import { testFile } from '../testUtil'
testFile('xx#pair.2.1.cram')
