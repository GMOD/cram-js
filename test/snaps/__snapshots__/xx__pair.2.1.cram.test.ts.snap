// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole xx#pair.2.1.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 2,
    "minorVersion": 1,
  },
  {
    "containerHeader": {
      "_endPosition": 39,
      "_size": 13,
      "alignmentSpan": 0,
      "landmarks": [
        0,
      ],
      "length": 10098,
      "numBases": 0,
      "numBlocks": 1,
      "numLandmarks": 1,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 39,
        "_size": 13,
        "alignmentSpan": 0,
        "landmarks": [
          0,
        ],
        "length": 10098,
        "numBases": 0,
        "numBlocks": 1,
        "numLandmarks": 1,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              0,
              10,
            ],
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              99,
              147,
            ],
          },
          "parametersBytes": 7,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              1,
              5,
            ],
          },
          "parametersBytes": 6,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              2,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              10,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 10153,
      "_size": 14,
      "alignmentSpan": 20,
      "landmarks": [
        144,
      ],
      "length": 271,
      "numBases": 60,
      "numBlocks": 5,
      "numLandmarks": 1,
      "numRecords": 6,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 10153,
        "_size": 14,
        "alignmentSpan": 20,
        "landmarks": [
          144,
        ],
        "length": 271,
        "numBases": 60,
        "numBlocks": 5,
        "numLandmarks": 1,
        "numRecords": 6,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 11,
              "readName": "a1",
              "sequenceId": 0,
              "uniqueId": 10306,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "a1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 10303,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 11,
              "readName": "b1",
              "sequenceId": 0,
              "uniqueId": 10307,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "b1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 10304,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 11,
              "readName": "c1",
              "sequenceId": 0,
              "uniqueId": 10308,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "c1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 10305,
          },
          {
            "alignmentStart": 11,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "a1",
              "sequenceId": 0,
              "uniqueId": 10303,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "a1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 10306,
          },
          {
            "alignmentStart": 11,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "b1",
              "sequenceId": 0,
              "uniqueId": 10304,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "b1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 10307,
          },
          {
            "alignmentStart": 11,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 10,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "c1",
              "sequenceId": 0,
              "uniqueId": 10305,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "c1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 10308,
          },
        ],
        "header": {
          "_endPosition": 10332,
          "_size": 30,
          "compressedSize": 30,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            20,
            6,
            0,
            3,
            2,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            187,
            244,
            222,
            109,
            132,
            151,
            161,
            25,
            221,
            166,
            224,
            116,
            82,
            22,
            67,
            220,
          ],
          "contentId": 0,
          "contentPosition": 10302,
          "contentType": "MAPPED_SLICE_HEADER",
          "parsedContent": {
            "_endPosition": 10183,
            "_size": 30,
            "contentIds": [
              11,
              12,
            ],
            "md5": [
              187,
              244,
              222,
              109,
              132,
              151,
              161,
              25,
              221,
              166,
              224,
              116,
              82,
              22,
              67,
              220,
            ],
            "numBlocks": 3,
            "numContentIds": 2,
            "numRecords": 6,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 20,
            "refSeqStart": 1,
          },
          "uncompressedSize": 30,
        },
      },
    ],
  },
]
`;
